File: 1480-running-sum-of-1d-array/running-sumv2.py
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        """
        This method creates a new list to store running sum using a more pythonic method with .append()

        Time: O(n) where N is the length of numbers
        Space: O(n) for the new list

        """
        # Create a new list to store the running sum. Initialise list with 0
        runningsums = []
        
        # Append the value from the 0th index from the original list to the new list
        runningsums.append(nums[0])
        # print(runningsums)

        # Iterate from the 1st index and do running sum. Append the running sum to the new list
        for i in range(1, len(nums)):
            runningsum = runningsums[i-1] + nums[i]
            runningsums.append(runningsum)
            # print(runningsums[i])

        # return new list 
        return runningsums

File: 1480-running-sum-of-1d-array/running-sumv2.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* runningSum(int* nums, int numsSize, int* returnSize){
        /*
        *This method uses existing list to store running sum
        *
        * Time: O(n) where N is the length of numbers
        * Space: O(1) for using existing list
        */  

        // numSize is store in *returnSize so that the value of numSize is not lost when this function is the popped off the stack after being called.
         *returnSize = numsSize;

        //Iterate from the 1st index and do running sum. Overwrite exist value with the running sum with the new value
    
        for (int i = 1; i < numsSize; i++)
        {
            int runningSum = nums[i] + nums[i-1];
            // printf("nums[i] %d\n", nums[i]);
            // printf("runningsum %d\n", runningSum);
            nums[i] = runningSum;
        }

        // Return new list
        return nums;
}
